src/events.js

```diff
@@ -316,28 +299,34 @@ function fireEvent(element, event) {
   return element.dispatchEvent(event)
 }
 
-Object.entries(eventMap).forEach(([key, {EventType = Event, defaultInit}]) => {
-  const eventName = key.toLowerCase()
+Object.entries(eventMap).forEach(
+  ([key, {EventType = 'Event', defaultInit}]) => {
+    const eventName = key.toLowerCase()
 
-  fireEvent[key] = (node, init) => {
-    const eventInit = {...defaultInit, ...init}
-    const {target: {value, files, ...targetProperties} = {}} = eventInit
-    Object.assign(node, targetProperties)
-    if (value !== undefined) {
-      setNativeValue(node, value)
-    }
-    if (files !== undefined) {
-      // input.files is a read-only property so this is not allowed:
-      // input.files = [file]
-      // so we have to use this workaround to set the property
-      Object.defineProperty(node, 'files', {
-        value: files,
-      })
+    fireEvent[key] = (node, init) => {
+      const eventInit = {...defaultInit, ...init}
+      const {target: {value, files, ...targetProperties} = {}} = eventInit
+      Object.assign(node, targetProperties)
+      if (value !== undefined) {
+        setNativeValue(node, value)
+      }
+      if (files !== undefined) {
+        // input.files is a read-only property so this is not allowed:
+        // input.files = [file]
+        // so we have to use this workaround to set the property
+        Object.defineProperty(node, 'files', {
+          value: files,
+        })
+      }
+      const window = node.ownerDocument.defaultView
+      if (typeof window[EventType] === 'undefined') {
+        window[EventType] = class extends window.Event {}
```

Comment:
Here's relevant line, btw. Previously this events were shimmed only in document.js helper